### ISSUE TITLE
Union support

### ISSUE THREAD

**Issue by mykola-mokhnach** [Koromix/rygel#63]
Hello,

I am currently trying the example https://koffi.dev/unions#win32-example and I have noticed that union support has been commented out from the types declaration: 

```
    export function struct(name: string, def: Record<string, TypeSpecWithAlignment>): IKoffiCType;
    export function struct(def: Record<string, TypeSpecWithAlignment>): IKoffiCType;
    export function pack(name: string, def: Record<string, TypeSpecWithAlignment>): IKoffiCType;
    export function pack(def: Record<string, TypeSpecWithAlignment>): IKoffiCType;

    // export function union(name: string, def: Record<string, TypeSpecWithAlignment>): IKoffiCType;
    // export function union(def: Record<string, TypeSpecWithAlignment>): IKoffiCType;
```

even though it is used in the actual example. Is there a reason for it? Can I still run the above example with the most recent version of the lib or it is necessary to downgrade?

**Comment by Koromix** [comment]
Honestly, I just forgot about it ><

I don't know the first thing about Typescript, I try to maintain the definition file since another user contributed it but in this case I forgot..

Will fix :)

**Comment by mykola-mokhnach** [comment]
so it is just necessary to uncomment the corresponding typedefs? And the union support itself is in a good shape?

**Comment by Koromix** [comment]
I probably need to change the typedefs, they seem incomplete.

The support for unions is recent, and was introduced in 2.5.0. It is in theory well supported but it is the most recent feature and so bugs are more likely.

I will fix these typedefs in Koffi 2.5.2.

**Comment by mykola-mokhnach** [comment]
I can confirm that the SendInput example works. Linking https://github.com/appium/appium-windows-driver/pull/179 for the reference.

Thanks for preparing good project documentation.

**Comment by Koromix** [comment]
Well, if anyone has any idea how to declare a function that you're supposed to call with new (for `new koffi.Union(type)`), I'd like to know 😁

**Comment by mykola-mokhnach** [comment]
Maybe @boneskull, our typing expert, could help there

**Comment by Koromix** [comment]
I think I got it, hopefully anyways, I've released Koffi 2.5.3 with an updated TS definition file :)
